.github/workflows/package-win.yml

```
@@ -42,6 +42,17 @@ jobs:
         shell: bash
         run: rm -rf dist dist-server release
       - run: pnpm package:win
+      # Optional Authenticode signing. Runs only when a cert is supplied via
+      # secrets.CSC_LINK (the same var electron-builder would use). Without it
+      # the build stays unsigned — the verify step below still rejects a
+      # publisherName on an unsigned build, so auto-update keeps working.
+      - name: sign the Windows build (optional)
+        if: ${{ secrets.CSC_LINK != '' }}
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)